WordPress/src/main/java/org/wordpress/android/ui/stockmedia/StockMediaPickerActivity.java

```diff
@@ -0,0 +1,590 @@
+package org.wordpress.android.ui.stockmedia;
+
+import android.app.FragmentManager;
+import android.os.Bundle;
+import android.os.Handler;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v7.app.ActionBar;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.GridLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.support.v7.widget.SearchView;
+import android.support.v7.widget.Toolbar;
+import android.text.Html;
+import android.text.TextUtils;
+import android.view.MenuItem;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.RelativeLayout;
+import android.widget.TextView;
+
+import org.greenrobot.eventbus.Subscribe;
+import org.greenrobot.eventbus.ThreadMode;
+import org.wordpress.android.R;
+import org.wordpress.android.WordPress;
+import org.wordpress.android.fluxc.Dispatcher;
+import org.wordpress.android.fluxc.generated.StockMediaActionBuilder;
+import org.wordpress.android.fluxc.model.SiteModel;
+import org.wordpress.android.fluxc.model.StockMediaModel;
+import org.wordpress.android.fluxc.store.StockMediaStore;
+import org.wordpress.android.ui.media.MediaPreviewActivity;
+import org.wordpress.android.util.ActivityUtils;
+import org.wordpress.android.util.AniUtils;
+import org.wordpress.android.util.AppLog;
+import org.wordpress.android.util.DisplayUtils;
+import org.wordpress.android.util.NetworkUtils;
+import org.wordpress.android.util.PhotonUtils;
+import org.wordpress.android.util.StringUtils;
+import org.wordpress.android.util.ToastUtils;
+import org.wordpress.android.util.WPLinkMovementMethod;
+import org.wordpress.android.widgets.WPNetworkImageView;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.inject.Inject;
+
+public class StockMediaPickerActivity extends AppCompatActivity implements SearchView.OnQueryTextListener {
+    private static final int MIN_SEARCH_QUERY_SIZE = 3;
+    private static final String TAG_RETAINED_FRAGMENT = "retained_fragment";
+
+    private static final String KEY_SEARCH_QUERY = "search_query";
+    private static final String KEY_CAN_LOAD_MORE = "can_load_more";
+    private static final String KEY_NEXT_PAGE = "next_page";
```

Comment:
Although it is unlikely to make a difference _right now_, there is a disconnection between the actual data and the `nextPage` and `canLoadMore` here. If the process were to be stopped, the data in the retained fragment would be destroyed, whereas these 3 states will be preserved. That can leave us in a state like the following:

```
mStockMediaList = new ArrayList<>(); //empty
mCanLoadMore = false;
mNextPage = 5;
```

It might be better to keep all 3 of them in the retained fragment and only save the search query in the saved state. We should set the search query in retained fragment from saved state and start a new search only if `searchQuery` in the fragment was empty but the saved state one is not (the process was stopped).